cns/restserver/homeazmonitor.go

```diff
@@ -50,14 +50,27 @@ func (h *HomeAzMonitor) updateCacheValue(resp cns.GetHomeAzResponse) {
 
 // readCacheValue reads home az cache value
 func (h *HomeAzMonitor) readCacheValue() cns.GetHomeAzResponse {
-	cachedResp, found := h.values.Get(homeAzCacheKey)
-	if !found {
-		return cns.GetHomeAzResponse{Response: cns.Response{
+	if cachedResp, found := h.values.Get(homeAzCacheKey); found {
+		return cachedResp.(cns.GetHomeAzResponse)
+	}
+
+	// Cache miss - populate cache and try again
+	ctx, cancel := context.WithTimeout(context.Background(), ContextTimeOut)
+	defer cancel()
+	h.Populate(ctx)
+
+	if cachedResp, found := h.values.Get(homeAzCacheKey); found {
+		return cachedResp.(cns.GetHomeAzResponse)
+	}
+
+	// Still no cache value after populate attempt
+	return cns.GetHomeAzResponse{
+		Response: cns.Response{
 			ReturnCode: types.NotFound,
```

Comment:
I think we should not return NotFound now. Initially we want to return NotFound when the cns start, but the request coming before the home az goroutine finish populating, and in that case, the cache is empty.  but now we know the populate function at least get execution once, then we can just return back cache values for GetHomeAzResponse.